SharedCode/src/commonMain/kotlin/ApplicationPresenter.kt

```diff
@@ -8,12 +8,34 @@ class ApplicationPresenter: ApplicationContract.Presenter() {
     private val dispatchers = AppDispatchersImpl()
     private var view: ApplicationContract.View? = null
     private val job: Job = SupervisorJob()
+    private val stations = listOf(
+        "BON",
+        "KGX",
+        "EUS",
+        "MAN",
+        "EDB"
+    )
+
+    private val stationSubmitButtonText = "Search"
 
     override val coroutineContext: CoroutineContext
         get() = dispatchers.main + job
 
     override fun onViewTaken(view: ApplicationContract.View) {
         this.view = view
-        view.setLabel(createApplicationScreenMessage())
+        view.populateOriginAndDestinationSpinners(stations)
+        view.setStationSubmitButtonText(stationSubmitButtonText)
+        view.setStationSubmitButtonHandler()
+    }
+
+    override fun onStationSubmitButtonPressed() {
+        // get inputs
+        var originStation = view?.getOriginStation()
+        var destinationStation = view?.getDestinationStation()
```

Comment:
Instead of having these gets, could we pass these two values through as parameters in the `onStationSubmitButtonPressed` function? Could possibly shorten this name to `onStationSubmitPressed` too

Reply:
Passed through as parameters now, and have removed the getters for the values selected with the spinners. Should we have kept those?